FiniteStateMachine/src/statemachine/State.java

```diff
@@ -2,11 +2,9 @@
 
 import org.jetbrains.annotations.NotNull;
 
-import java.util.EnumSet;
+import java.util.Set;
 
 
-public interface State<E extends Enum<E> & State<E>> {
-    @NotNull EnumSet<E> nextStates();
```

Comment:
True, I could use `Set<E>` here, and have the `Gear` implementation be more specific.